src/linear_algebra/free_module/ideal_quotient.lean

```diff
@@ -9,6 +9,7 @@ import linear_algebra.free_module.finite.rank
 import linear_algebra.free_module.pid
 import linear_algebra.free_module.strong_rank_condition
 import linear_algebra.quotient_pi
+import ring_theory.norm
```

Comment:
I think this import is going to be problematic, for a few reasons ordered from weak to strong:
 * I don't like adding imports to existing files in general
 * `ring_theory.norm` adds quite a few transitive dependencies compared to this file
 * `ring_theory.norm` is far away from being ported to mathlib4. So it would not be possible to forward-port the changes you made to this file.

Could you move the new definitions to a new file? (Not sure about the name: `linear_algebra.free_module.norm`? `ring_theory.norm.free_module_quotient`?)